### PR TITLE
Deprecate vectorized clamp methods in favor of compact broadcast syntax

### PR DESCRIPTION
**Pull request by Sacha0**
This PR deprecates all remaining vectorized `clamp` methods in favor of compact broadcast syntax. Ref. #16285, #17302, #18495, #18512, #18513, #18558, #18564, #18566, #18571 #18575, #18576, #18586, #18590, #18593, #18607, and #18608. Best!

(Unlike with `float`, `real`, etc., the remaining vectorized `clamp` methods never alias their input. This PR should be less controversial than #18495, #18512, and #18513 as a result.)

### CODE REVIEW

File: base/math.jl [hunk 1]
```diff
@@ -38,10 +38,10 @@ import Core.Intrinsics: sqrt_llvm, box, unbox, powi_llvm
     clamp(x, lo, hi)
 
 Return `x` if `lo <= x <= hi`. If `x < lo`, return `lo`. If `x > hi`, return `hi`. Arguments
-are promoted to a common type. Operates elementwise over `x` if `x` is an array.
```
Comment (tkelman) [inline base/math.jl]: does this need a rerun of `make docs`?

File: test/offsetarray.jl [hunk 2]
```diff
@@ -313,7 +313,7 @@ seek(io, 0)
 @test readdlm(io, eltype(A)) == parent(A)
 
 amin, amax = extrema(parent(A))
-@test clamp(A, (amax+amin)/2, amax) == clamp(parent(A), (amax+amin)/2, amax)
+@test clamp.(A, (amax+amin)/2, amax).parent == clamp.(parent(A), (amax+amin)/2, amax)
```
Comment (tkelman) [inline test/offsetarray.jl]: why does this need `.parent` now?

File: base/math.jl (hunk 1 above)
Comment (Sacha0) [inline base/math.jl]: That it does! Good catch. Fixed. Thanks!

File: test/offsetarray.jl (hunk 2 above)
Comment (Sacha0) [inline test/offsetarray.jl]: The operation on the left produces a `TestHelpers.OAs.OffsetArray` (from `test/TestHelpers.jl`) whereas that on the right produces an `Array`. Comparing an `OffsetArray` to an `Array` seems to fail even when their contents are identical, hence the `.parent`. Thoughts? Thanks!

Comment (tkelman) [inline test/offsetarray.jl]: and without broadcast what type did it return?

Comment (TotalVerb) [inline test/offsetarray.jl]: An `OffsetArray` ought not to be substituted for an `Array`. If `broadcast` is not preserving offsets, then that needs to be fixed.

Comment (Sacha0) [inline test/offsetarray.jl]: The existing vectorized `clamp` methods return an `Array` subtype independent of the input type (for example, `clamp`ing a `SparseMatrixCSC` or `OffsetArray` yields an `Array`). On the other hand `broadcast` better preserves the input type (for example, `clamp`ing a `SparseMatrixCSC` yields a `SparseMatrixCSC` and the analog holds for `OffsetArray`s). Thoughts? Thanks!

Comment (tkelman) [inline test/offsetarray.jl]: Oh, interesting, guess this is a bugfix then. Should there be a few more tests of this behavior?

Comment (TotalVerb) [inline test/offsetarray.jl]: Since `clamp` is not zero-preserving, I think the old behaviour was preferred.

Comment (tkelman) [inline test/offsetarray.jl]: oh right, only zero-preserving if the bounds include the origin, but sometimes preserving structure wouldn't be type stable

Comment (Sacha0) [inline test/offsetarray.jl]: Good catch! Tricky. Special case sparse and structured matrix types?

**Comment by Sacha0** [comment]
Rebased. Best!

File: test/offsetarray.jl (hunk 2 above)
Comment (tkelman) [inline test/offsetarray.jl]: the broadcast code will now check if this is zero preserving right? and return dense-in-sparse-format if not?

Comment (Sacha0) [inline test/offsetarray.jl]: Should, yes. Best!

**Comment by Sacha0** [comment]
Subsumed by #19791.

**Pull request closed without merging**